### PR TITLE
Logging drops

### PR DESCRIPTION
**Pull request by boaks**
Add logger for message drops.

Signed-off-by: Achim Kraus <achim.kraus@bosch.io>

### CODE REVIEW

File: scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java [hunk 1]
```diff
@@ -2351,7 +2352,7 @@ private void sendMessage(final RawData message, final Connection connection, fin
 		} catch (IOException e) {
 			message.onError(e);
 		} catch (GeneralSecurityException e) {
-			LOGGER.debug("Cannot send APPLICATION record to peer [{}]", message.getInetSocketAddress(), e);
+			DROP_LOGGER.debug("Cannot send APPLICATION record to peer [{}]", message.getInetSocketAddress(), e);
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java:2355]: It seems to me this is more than a `Debug` ? maybe a `warn`?

File: scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java [hunk 2]
```diff
@@ -1151,10 +1152,10 @@ public void run() {
 				health.receivingRecord(true);
 			}
 			if (connectionId == null) {
-				LOGGER.debug("Discarding {} records from [{}] received without existing connection",
+				DROP_LOGGER.debug("Discarding {} records from [{}] received without existing connection",
```
[the remark this answers is not in the document]
Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java:1155]: OK.

File: scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java (hunk 1 above)
Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java:2355]: OK

**Comment by boaks** [comment]
> But I'm not sure to see why those specific logs are in INFO ? You consider address changed or identity mismatched as upper issue ?

That depends on your usage. With CLOSE_NOTIFY and server initiated requests, yes. For those, who use it mainly as "coap-server", that may indicate some implementation issues.

**Comment by boaks** [comment]
And just to mention:

I assume, we are not really ready with the "CLOSE_NOTIFIES".

**Comment by sbernard31** [comment]
LGTM

**Comment by sbernard31** [comment]
Could I cherry pick this for the 2.2.x branch ?

**Comment by boaks** [comment]
sure

**Pull request merged**